Sources/TuistDependencies/SwiftPackageManager/SwiftPackageManagerInteractor.swift

```diff
@@ -0,0 +1,190 @@
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+enum SwiftPackageManagerInteractorError: FatalError, Equatable {
+    /// Thrown when `Package.resolved` cannont be found in temporary directory after `Swift Package Manager` installation.
+    case packageResolvedNotFound
+    /// Thrown when `.build` directory cannont be found in temporary directory after `Swift Package Manager` installation.
+    case buildDirectoryNotFound
+
+    /// Error type.
+    var type: ErrorType {
+        switch self {
+        case .packageResolvedNotFound,
+             .buildDirectoryNotFound:
+            return .abort
+        }
+    }
+
+    /// Error description.
+    var description: String {
+        switch self {
+        case .packageResolvedNotFound:
+            return "Package.resolved file was not found after Carthage installation."
+        case .buildDirectoryNotFound:
+            return ".build directory was not found after Carthage installation."
+        }
+    }
+}
+
+// MARK: - Swift Package Manager Interacting
+
+public protocol SwiftPackageManagerInteracting {
+    /// Fetches `Swift Package Manager` dependencies.
+    /// - Parameter dependenciesDirectory: The path to the directory that contains the `Tuist/Dependencies/` directory.
+    /// - Parameter dependencies: List of dependencies to intall using `Swift Package Manager`.
+    func fetch(dependenciesDirectory: AbsolutePath, dependencies: SwiftPackageManagerDependencies) throws
+}
+
+// MARK: - Swift Package Manager Interactor
+
+public final class SwiftPackageManagerInteractor: SwiftPackageManagerInteracting {
+    private let fileHandler: FileHandling
+
+    public init(
+        fileHandler: FileHandling = FileHandler.shared
```

Comment:
I don't really think we need to be injecting `FileHandler` manually as every `TuistUnitTestCase`should inject `MockFileHandler` by default. You can just use `FileHandler.shared` directly